src/backend/distributed/commands/alter_table.c

```diff
@@ -360,6 +360,54 @@ worker_change_sequence_dependency(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * UndistributeTables undistributes given relations. It first collects all foreign keys
+ * to recreate them after the undistribution. Then, drops the foreign keys and
+ * undistributes the relations. Finally, it recreates foreign keys.
+ */
+void
+UndistributeTables(List *relationIds)
+{
+	List *tablesToConvert = NIL;
+	Oid relationId = InvalidOid;
+	foreach_oid(relationId, relationIds)
+	{
+		/*
+		 * Skip partitions as they would be distributed/undistibuted by partitioned table.
+		 *
+		 * We should filter out partitions here before calling CreateTenantSchemaTable.
+		 * Otherwise, converted partitioned table would change oid of partitions and its
+		 * partition tables would fail wih oid not exist.
+		 */
+		if (PartitionTable(relationId))
+		{
+			continue;
+		}
```

Comment:
Should the callers filter those tables instead? Because I wouldn't expect a function named as such to skip some of the input tables. This could be misleading for the future callers that want to undistribute a parent and a child table that are not related to each other, as in:

CREATE PARTITONED TABLE p1;
CREATE TABLE p1_c1 PARTITON OF p1:

CREATE PARTITONED TABLE p2;

UndistributeTables(list_make_2(p1_c1, c2));

I also don't think a future caller would ever need to do such a thing but to me it sounds safer to do that in caller site.

Reply:
makes sense